features/step_definitions/login_and_dashboard_steps.rb

```diff
@@ -1,3 +1,30 @@
+When(/^I request access using my e\-mail$/) do
+  page = UI::Pages::Login.new
+  page.load
+
+  page.email_field.set @user.email
+  page.request_button.click
```

Comment:
I think you can wrap the whole thing to a helper `login_as` where you can provide correct / incorrect e-mail as a parameter. Then step on line 16 can reuse completely.